### PR TITLE
[orchestration] add apigroupversion tag

### PR DESCRIPTION
**Pull request by nammn**
### What does this PR do?

- adding the `kube_api_version` tag for the orchestrator explorer 
- By re-using the `extraTags` we can make the change in one-place and affect all the resources
Screenshot 2022-11-09 at 15 23 55
Screenshot 2022-11-09 at 15 24 14

### Motivation

- adding the `kube_api_version` tag for the orchestrator explorer because grouping and filtering for them can be useful (think about version migrations)

### CODE REVIEW

File: pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go [hunk 1]
```diff
@@ -105,3 +105,15 @@ type CollectorRunResult struct {
 	ResourcesListed    int
 	ResourcesProcessed int
 }
+
+func GetProcessorContext(rcfg *CollectorRunConfig, metadata *CollectorMetadata) *processors.ProcessorContext {
```
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go:109]: Oh yeah indeed !

File: pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go [hunk 2]
```diff
@@ -223,6 +223,7 @@ func (cb *CollectorBundle) Run(sender aggregator.Sender) {
 		runStartTime := time.Now()
 
 		result, err := collector.Run(cb.runCfg)
+
```
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go]: yes, during some refactoring 😮‍💨 
Fixed in https://github.com/DataDog/datadog-agent/pull/14239/commits/da80d2ce5378b4c0a5588c95be70b448a12c5e3b

File: pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go [hunk 3]
```diff
@@ -105,3 +105,15 @@ type CollectorRunResult struct {
...
+	ctx := &processors.ProcessorContext{
+		APIClient:       rcfg.APIClient,
+		Cfg:             rcfg.Config,
+		ClusterID:       rcfg.ClusterID,
+		MsgGroupID:      rcfg.MsgGroupRef.Inc(),
+		NodeType:        metadata.NodeType,
+		ApiGroupVersion: metadata.Version,
```
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go]: Fixed in https://github.com/DataDog/datadog-agent/pull/14239/commits/da80d2ce5378b4c0a5588c95be70b448a12c5e3b

File: pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go]: Fixed in https://github.com/DataDog/datadog-agent/pull/14239/commits/da80d2ce5378b4c0a5588c95be70b448a12c5e3b

File: pkg/collector/corechecks/cluster/orchestrator/processors/k8s/clusterrole.go [hunk 4]
```diff
@@ -46,8 +48,7 @@ func (h *ClusterRoleHandlers) BuildMessageBody(ctx *processors.ProcessorContext,
 		GroupId:      ctx.MsgGroupID,
 		GroupSize:    int32(groupSize),
 		ClusterRoles: models,
-		Tags:         ctx.Cfg.ExtraTags,
-	}
+		Tags:         append(ctx.Cfg.ExtraTags, fmt.Sprintf("%s:%s", "kube_api_version", ctx.ApiGroupVersion))}
```
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/collector/corechecks/cluster/orchestrator/processors/k8s/clusterrole.go]: Fixed in https://github.com/DataDog/datadog-agent/pull/14239/commits/da80d2ce5378b4c0a5588c95be70b448a12c5e3b

**Review by maycmlee — APPROVED** [review]
👍 from docs.

**Review by xlucas — APPROVED** [review]

**Pull request merged**